src/workspace/Lesson/SectionOverview/Preview/index.jsx

```diff
@@ -1,87 +1,184 @@
-            el => el._id === store.view.activeSectionId,
-        )
-
-		// DnD
-		const wrapperRef = useRef(null)
-		const previewRef = useRef(null)
-		if (connectDragSource) {
-			connectDragPreview(previewRef)
-			connectDropTarget(wrapperRef)
-			// const opacity = isDragging ? 0 : 1
-			useImperativeHandle(ref, () => ({
-				getNode: () => wrapperRef.current,
-			}))
-		}
-
-        const p = (
-            <div>
-                <OuterMost noWrap ref={wrapperRef} delete={section.delete}>
-                    <Controls
-                        connectDragSource={connectDragSource}
-                        sectionId={section._id}
-                        store={store}
-                        index={index}
-                        dispatch={dispatch}
-                        visible={section.visible}
-                        sectionTitle={section.title}
-                    />
-                    <EditorPreview
-                        store={store}
-                        section={section}
-                        expanded={expanded}
-                        index={index}
-                        connectDragSource={connectDragSource}
-                        activeSectionIndex={activeSectionIndex}
-                        dispatch={dispatch}
-                        wrapperRef={wrapperRef}
-                        previewRef={previewRef}
-                        isDragging={isDragging}
-                        k={k}
-                    />
-                </OuterMost>
-            </div>
-        )
-        return connectDragSource(p)
-    },
-)
-
-export default DnDHOC(Preview)
+const Wrapper = styled.div`
+    overflow: hidden;
+    cursor: pointer;
+    box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.4);
+    transition: 250ms all ease-in-out;
+    position: relative;
+    font-size: 20px;
+    line-height: 1.42;
+    height: 100%;
+	margin: 15px auto;
+	width: 200px;
+	height: 160px;
+
+    ${props =>
+		props.expanded
+			? css`
+					margin: 8px 10px;
+					border-radius: 3px;
+					background-color: #fff;
+					border: 3px solid rgba(68, 68, 68, 1);
+					${props =>
+		!props.active &&
+						css`
+							border-color: #fff;
+						`}
+			  `
+			: css`
+					border: 3px solid transparent;
+					background-color: rgba(170, 170, 170, 1);
+					width: 18px;
+					height: 18px;
+					border-radius: 9px;
+					box-shadow: none !important;
+					transform: scale(0.5);
+					transform-origin: center center;
+
+					${props =>
+		props.active &&
+						css`
+							transform: scale(1);
+						`}
+
+					&:hover {
+						transform: scale(1);
+					}
+			  `}
+
+    &:hover {
+        box-shadow: 0 3px 10px 0 rgba(0, 0, 0, 0.6);
+    }
+
+    ${({ visible }) => {
+		return (
+			!visible &&
+			css`
+				opacity: 0.7;
+			`
+		)
+	}}
+
+    display: ${({ hidden }) => hidden && "none"};
+
+`
+
+const Preview = ({
+	store,
+	active,
+	section,
+	expanded,
+	index,
+	activeSectionIndex,
+	dispatch,
+	k,
+	getSectionIndex,
```

Comment:
Ich finde nicht das es Aufgabe des Preview ist seine Position zu verwalten, oder zu updaten. Bzw. überhaupt zu wissen, das er Teil einer Collection ist. 
Daher activeSectionIndex, getSectionIndex, moveSection sollten nicht Teil dieser Komponenten sein.
Das gleiche gilt für den drop handler. 
dragable Komponenten und drop zone sollten durch eine Wrapper Komponenten verwaltet werden, nicht als eingebauter Teil der Komponente.

Wofür wird das k hier verwendet?

Reply:
das k keine ahnung ist noch von lukas. Das drag and drop kann die komponente doch ruhig initalisieren. viel mehr macht die hier eh nicht. Wir können uns irgendwann überlegen ob man das nicht rausziehen kann oder ob es sinn ergibt aber vorerst haben wir das nur an der einen stelle und ich sehe auch keinen weiteren anwendungsfall für diese art des drag and drops weswegen ich komplexität sparen wollte. Das was lukas gemacht hatte ist inzwischen als veraltet markiert, weswegen ich es gleich neu gebaut hab statt das alte anzupassen. Active section index kann an sich raus, weis auch gar nicht ob das noch verwendet wird, der rest ist fürs drag an drop notwendig

Reply:
k, store index und so ist rausgeflogen aber die funktionen für das drag and drop würde ich drin lassen. Die Section kennt ihr position im array dann ja nicht sondern verwendet nur die funtkionen

Reply:
ticket ist erstellt